cmd/tink-worker/internal/registry.go

```diff
@@ -0,0 +1,75 @@
+package internal
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"os"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var log *logrus.Logger
+
+type RegistryConnDetails struct {
+	registry,
+	user,
+	pwd string
+	logger *logrus.Logger
+}
+
+func SetLogger(logger *logrus.Logger) {
+	log = logger
+}
+
+func NewRegistryConnDetails(registry, user, pwd string, logger *logrus.Logger) *RegistryConnDetails {
+	return &RegistryConnDetails{
+		registry: registry,
+		user:     user,
+		pwd:      pwd,
+		logger:   logger,
+	}
+}
+
+func (r *RegistryConnDetails) NewClient() (*client.Client, error) {
+	if r.registry == "" {
+		return nil, errors.New("required DOCKER_REGISTRY")
+	}
+	c, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+
+	if err != nil {
+		return nil, errors.Wrap(err, "DOCKER CLIENT")
+	}
+
+	r.logger.SetFormatter(&logrus.JSONFormatter{})
+	return c, nil
+}
+
+// pullImage outputs to stdout the contents of the requested image (relative to the registry)
+func (w *RegistryConnDetails) pullImage(ctx context.Context, cli *client.Client, image string) error {
+	registry := w.registry
```

Comment:
we should either use `registry` in line 58 or `w.registry` on line 66.

Reply:
I think this was some attempt to keep my initial diff noise down between local refactors. What a mess.